### PR TITLE
Anoncreds Revocation NodeJs sample

### PR DESCRIPTION
**Pull request by Artemkaaas**
Signed-off-by: artem.ivanov <artem.ivanov@dsr-company.com>

### CODE REVIEW

File: samples/nodejs/doc/anoncredsRevocationScenario.puml [hunk 1]
```diff
@@ -29,8 +29,8 @@ note over I, V
 Issuer share public data (schema ID, credential definition, ...)
 end note
 
-P -> L : Gets schema from ledger
-V -> L : Gets schema from ledger
+P <- L : Gets schema from ledger
```
[the remark this answers is not in the document]
Comment (Artemkaaas) [reply samples/nodejs/doc/anoncredsRevocationScenario.puml]: Ok. I agree.

File: samples/nodejs/src/anoncredsRevocationScenario.js [hunk 2]
```diff
@@ -315,6 +322,7 @@ async function run() {
 
     log("Transfer credential from 'Issuer' to 'Prover' (via HTTP or other) ...")
     prover.cred = issuer.cred
+    issuer.cred = undefined
```
[the remark this answers is not in the document]
Comment (Artemkaaas) [reply samples/nodejs/src/anoncredsRevocationScenario.js]: It's an optional step that not relates to the revocation process directly.
I added it to show that in an ideal way Issuer shouldn't store user credentials.

File: samples/nodejs/src/anoncredsRevocationScenario.js [hunk 3]
```diff
@@ -331,15 +339,15 @@ async function run() {
         "requested_attributes": {
             "attr1_referent": {
                 "name": "name",
-                "restrictions": [{"cred_def_id": verifier.credDefId}]
+                "restrictions": {"cred_def_id": verifier.credDefId}
```
[the remark this answers is not in the document]
Comment (Artemkaaas) [reply samples/nodejs/src/anoncredsRevocationScenario.js]: Yes. The format you used is outdated. 
However, Libindy supports both this and  [wql](https://github.com/hyperledger/indy-sdk/tree/master/doc/design/011-wallet-query-languagel) 
wql is more flexible and is similar to mongo style queries
(here)[https://github.com/hyperledger/indy-sdk/blob/master/libindy/tests/anoncreds.rs#L2476] are examples

File: samples/nodejs/src/anoncredsRevocationScenario.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (xaviernopre) [reply samples/nodejs/src/anoncredsRevocationScenario.js]: OK 👍

File: samples/nodejs/src/anoncredsRevocationScenario.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (xaviernopre) [reply samples/nodejs/src/anoncredsRevocationScenario.js]: OK, thanks

File: samples/nodejs/src/anoncredsRevocationScenario.js [hunk 4]
```diff
@@ -296,15 +303,15 @@ async function run() {
     }
 
     logIssuer("Issuer creates credential")
-    issuer.credValues = {
+    credValues = {
```
[the remark this answers is not in the document]
Comment (xaviernopre) [reply samples/nodejs/src/anoncredsRevocationScenario.js]: You need to add `const` before the variable ;-)

**Pull request merged**